src/analysis/calculus/times_cont_diff.lean

```diff
@@ -1210,6 +1210,10 @@ lemma times_cont_diff_within_at.times_cont_diff_at {n : with_top ℕ}
   times_cont_diff_at 𝕜 n f x :=
 by rwa [times_cont_diff_at, ← times_cont_diff_within_at_inter hx, univ_inter]
 
+lemma times_cont_diff_on_of_times_cont_diff_at {n : with_top ℕ}
+  (h : ∀ x ∈ s, times_cont_diff_at 𝕜 n f x) : times_cont_diff_on 𝕜 n f s :=
+λ x hx, times_cont_diff_at.times_cont_diff_within_at (h x hx)
```

Comment:
```suggestion
λ x hx, (h x hx).times_cont_diff_within_at 
```

Reply:
I think you still have to give it a name to apply it and change the goal. I mean all the lemmas that are proven with `rfl` have a proof that is shorter than the lemma name, but they are still useful!

Reply:
Still, I think it is better not to add a debatable lemma like this until one has a definite use case.  Note that there is no `differentiable_on_of_differentiable_at` or `continuous_on_of_continuous_at`, although those APIs have been tested more extensively than the `times_cont_diff` API.